grayskull/pypi/pypi.py

```diff
@@ -59,21 +59,36 @@ def __init__(
         self["build"]["script"] = "<{ PYTHON }} -m pip install . -vv"
 
     @staticmethod
-    def _download_sdist_pkg(sdist_url: str, dest: str):
+    def _pkg_name_from_sdist_url(sdist_url: str):
+        if origin_is_github(sdist_url):
+            return sdist_url.split("/")[-3] + ".tar.gz"
+        else:
+            return sdist_url.split("/")[-1]
+
+    @staticmethod
+    def _generate_git_archive_tarball_url(git_url: str) -> str:
+        """This method takes a github repository url and returns the archive
+        tarball url for that repository.
+        :param git_url: github repository url
+        :return: github repository archive tarball url
+        """
+        archive_tarball_url = git_url + "/archive/main.tar.gz"
```

Comment:
Let's add the `git_ref` keyword argument here already, with a default value of `main`:

```suggestion
        archive_tarball_url = git_url + f"/archive/{git_ref}.tar.gz"
```